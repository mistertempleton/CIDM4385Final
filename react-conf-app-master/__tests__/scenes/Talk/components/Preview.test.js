// @flow
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Preview from '../../../../app/scenes/Talk/components/Preview';

describe('Talk - Preview', () => {
  it('renders correctly in top position', () => {
    const tree = renderer
      .create(
        <Preview
          position="top"
          speakerName="Max Stoiber"
          talkStartTime="9AM"
          title="React Boilerplate"
          subtitle="React Boilerplate"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly in bottom position', () => {
    const tree = renderer
      .create(
        <Preview
          position="bottom"
          speakerName="Max Stoiber"
          talkStartTime="9AM"
          title="React Boilerplate"
          subtitle="React Boilerplate"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
