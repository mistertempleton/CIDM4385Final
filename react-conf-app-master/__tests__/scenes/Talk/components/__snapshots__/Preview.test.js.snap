exports[`Talk - Preview renders correctly in bottom position 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 70,
        "justifyContent": "center",
        "paddingHorizontal": 60,
        "width": 750,
      },
      Object {
        "left": 0,
        "position": "absolute",
      },
      Object {
        "bottom": -70,
      },
    ]
  }>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "textAlign": "center",
      }
    }>
    React Boilerplate
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#666",
        "textAlign": "center",
      }
    }>
    React Boilerplate
  </Text>
</View>
`;

exports[`Talk - Preview renders correctly in top position 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 70,
        "justifyContent": "center",
        "paddingHorizontal": 60,
        "width": 750,
      },
      Object {
        "left": 0,
        "position": "absolute",
      },
      Object {
        "top": -70,
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "textAlign": "center",
      }
    }>
    React Boilerplate
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#666",
        "textAlign": "center",
      }
    }>
    React Boilerplate
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#383838",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;
